File: app/src/main/java/com/tj/newsreader/di/ViewModelFactoryModule.java
package com.tj.newsreader.di;


import androidx.lifecycle.ViewModelProvider;

import com.tj.newsreader.viewmodels.ViewModelProviderFactory;

import dagger.Binds;
import dagger.Module;


//responsible for doing dependency injection in the model class
@Module
public abstract class ViewModelFactoryModule {
    //Injects into MainActivity
    @Binds
    public abstract ViewModelProvider.Factory bindViewModelFactory(ViewModelProviderFactory modelProviderFactory);
}

File: app/src/main/java/com/tj/newsreader/di/ViewModelFactoryModule.kt
package com.tj.newsreader.di


import androidx.lifecycle.ViewModelProvider

import com.tj.newsreader.viewmodels.ViewModelProviderFactory

import dagger.Binds
import dagger.Module


//responsible for doing dependency injection in the model class
@Module
abstract class ViewModelFactoryModule {
    //Injects into MainActivity
    @Binds
    abstract fun bindViewModelFactory(modelProviderFactory: ViewModelProviderFactory): ViewModelProvider.Factory
}
